packages/components/src/Datalist/Datalist.component.js

```diff
@@ -315,6 +315,7 @@ class Datalist extends Component {
 					readOnly={this.props.readOnly || false}
 					theme={this.theme}
 					value={this.state.value}
+					noResultText={this.props.noResultText}
```

Comment:
please spread props here and remove autoFocus, disabled, ...

Reply:
As discussed spreading the props would pass unwanted/unecessary one to the typeahead underneath so having this props whitelist is more suited. In addition we don't want to pass all props that the typeahead can manage cause the datalist is a specific usage of the typeahead.